webui/react/src/pages/ModelRegistry.tsx

```diff
@@ -0,0 +1,404 @@
+
+  const handleNameSearchApply = useCallback((newSearch: string) => {
+    updateSettings({ name: newSearch || undefined });
+  }, [ updateSettings ]);
+
+  const handleNameSearchReset = useCallback(() => {
+    updateSettings({ name: undefined });
+  }, [ updateSettings ]);
+
+  const nameFilterSearch = useCallback((filterProps: FilterDropdownProps) => (
+    <TableFilterSearch
+      {...filterProps}
+      value={settings.name || ''}
+      onReset={handleNameSearchReset}
+      onSearch={handleNameSearchApply}
+    />
+  ), [ handleNameSearchApply, handleNameSearchReset, settings.name ]);
+
+  const handleDescriptionSearchApply = useCallback((newSearch: string) => {
+    updateSettings({ description: newSearch || undefined });
+  }, [ updateSettings ]);
+
+  const handleDescriptionSearchReset = useCallback(() => {
+    updateSettings({ description: undefined });
+  }, [ updateSettings ]);
+
+  const descriptionFilterSearch = useCallback((filterProps: FilterDropdownProps) => (
+    <TableFilterSearch
+      {...filterProps}
+      value={settings.description || ''}
+      onReset={handleDescriptionSearchReset}
+      onSearch={handleDescriptionSearchApply}
+    />
+  ), [ handleDescriptionSearchApply, handleDescriptionSearchReset, settings.description ]);
+
+  const handleLabelFilterApply = useCallback((tags: string[]) => {
+    updateSettings({ tags: tags.length !== 0 ? tags : undefined });
+  }, [ updateSettings ]);
+
+  const handleLabelFilterReset = useCallback(() => {
+    updateSettings({ tags: undefined });
+  }, [ updateSettings ]);
+
+  const labelFilterDropdown = useCallback((filterProps: FilterDropdownProps) => (
+    <TableFilterDropdown
+      {...filterProps}
+      multiple
+      searchable
+      values={settings.tags}
+      onFilter={handleLabelFilterApply}
+      onReset={handleLabelFilterReset}
+    />
+  ), [ handleLabelFilterApply, handleLabelFilterReset, settings.tags ]);
+
+  const showConfirmDelete = useCallback((model: ModelItem) => {
+    Modal.confirm({
+      closable: true,
+      content: `Are you sure you want to delete this model "${model.name}" and all 
+      of its versions from the model registry?`,
+      icon: null,
+      maskClosable: true,
+      okText: 'Delete Model',
+      okType: 'danger',
+      onOk: () => deleteCurrentModel(model),
+      title: 'Confirm Delete',
+    });
+  }, [ deleteCurrentModel ]);
+
+  const columns = useMemo(() => {
+    const labelsRenderer = (value: string, record: ModelItem) => (
+      <TagList
+        compact
+        tags={record.labels ?? []}
+        onChange={(tags) => setModelTags(record.id, tags)}
+      />
+    );
+
+    const overflowRenderer = (_:string, record: ModelItem) => {
+      const isDeletable = user?.isAdmin || user?.username === record.username;
+      return (
+        <Dropdown
+          overlay={(
+            <Menu>
+              <Menu.Item
+                key="switch-archived"
+                onClick={() => switchArchived(record)}>
```

Comment:
**chore:** Archiving and unarchiving can use a `try/catch` block plus `async/await` to ensure the right sequence of API call + UI update.